src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -53,19 +48,26 @@ public async Task<InstanceDiscoveryMetadataEntry> GetMetadataAsync(Uri authority
             else
             {
                 logger.Verbose($"[Region Discovery] The network provider found an entry for {environment}");
+                LogTelemetryData(cachedEntry.PreferredNetwork.Split('.')[0], RegionSource.Cache, requestContext);
             }
-
-            requestContext.ApiEvent.RegionDiscovered = cachedEntry.PreferredNetwork.Split('.')[0];
+            
             return cachedEntry;
         }
 
 
-        private async Task<string> GetRegionAsync(ICoreLogger logger)
+        private async Task<string> GetRegionAsync(RequestContext requestContext)
         {
+            ICoreLogger logger = requestContext.Logger;
+            string region;
+
             if (!Environment.GetEnvironmentVariable(RegionName).IsNullOrEmpty())
```

Comment:
Don't we want to extract the region in a variable before the test? isn't this a bit costly to call GetEnvironmentVariable(RegionName) twice ?